File: .ghidra/search_strings.py
# Search for protocol-related strings in DCX-Remote.exe
# @category Analysis

from ghidra.program.model.listing import CodeUnit

# Get the current program
program = currentProgram
listing = program.getListing()

# Keywords to search for
keywords = ["XPCR", "XPRB", "XSNP", "XCUR", "preset", "Preset", "PRESET", 
            "save", "Save", "SAVE", "copy", "Copy", "COPY", 
            "edit", "Edit", "EDIT", "buffer", "Buffer", "BUFFER",
            ".dcx", "DCX", "offset", "Offset"]

print("Searching for protocol-related strings...")
print("=" * 80)

# Search through all defined strings
string_iter = listing.getDefinedData(True)
found_strings = []

for data in string_iter:
    if data.hasStringValue():
        string_value = data.getValue()
        if string_value:
            string_text = str(string_value)
            for keyword in keywords:
                if keyword in string_text:
                    addr = data.getAddress()
                    found_strings.append((addr, string_text, keyword))
                    print("Found '{}' at {}: {}".format(keyword, addr, string_text[:100]))
                    break

print("=" * 80)
print("Total strings found: {}".format(len(found_strings)))

File: .ghidra/SearchStrings.java
// Search for protocol-related strings in DCX-Remote.exe
// @category Analysis

import ghidra.app.script.GhidraScript;
import ghidra.program.model.listing.*;
import ghidra.program.model.address.*;

public class SearchStrings extends GhidraScript {
    
    @Override
    public void run() throws Exception {
        // Keywords to search for
        String[] keywords = {
            "XPCR", "XPRB", "XSNP", "XCUR", 
            "preset", "Preset", "PRESET",
            "save", "Save", "SAVE",
            "copy", "Copy", "COPY",
            "edit", "Edit", "EDIT",
            "buffer", "Buffer", "BUFFER",
            ".dcx", "DCX",
            "offset", "Offset", "OFFSET"
        };
        
        println("Searching for protocol-related strings...");
        println("================================================================================");
        
        Listing listing = currentProgram.getListing();
        AddressSetView dataAddresses = currentProgram.getMemory().getLoadedAndInitializedAddressSet();
        
        int foundCount = 0;
        
        // Iterate through all defined data
        DataIterator dataIter = listing.getDefinedData(dataAddresses, true);
        while (dataIter.hasNext() && !monitor.isCancelled()) {
            Data data = dataIter.next();
            
            if (data.hasStringValue()) {
                String stringValue = (String) data.getValue();
                if (stringValue != null) {
                    for (String keyword : keywords) {
                        if (stringValue.contains(keyword)) {
                            Address addr = data.getAddress();
                            String preview = stringValue.length() > 100 ? 
                                stringValue.substring(0, 100) + "..." : stringValue;
                            println(String.format("Found '%s' at %s: %s", keyword, addr, preview));
                            foundCount++;
                            break;
                        }
                    }
                }
            }
        }
        
        println("================================================================================");
        println(String.format("Total strings found: %d", foundCount));
    }
}
